test/unit/test_borealis.py

```diff
@@ -0,0 +1,398 @@
+        given data
+
+        Expected Behaviour
+        -------------------
+        Raises BorealisDataFormatTypeError because the bfiq data has the
+        wrong type for the first_range_rtt field
+        """
+        bfiq_incorrect_fmt = copy.deepcopy(
+            borealis_bfiq_data_sets.borealis_bfiq_data)
+        keys = sorted(list(bfiq_incorrect_fmt.keys()))
+        bfiq_incorrect_fmt[keys[0]]['first_range_rtt'] = 5
+        writer = pydarn.BorealisWrite(
+            "test_bfiq.bfiq.hdf5", bfiq_incorrect_fmt)
+
+        try:
+            writer.write_bfiq()
+        except pydarn.borealis_exceptions.BorealisDataFormatTypeError as err:
+            self.assertEqual(
+                err.incorrect_params['first_range_rtt'], "<class 'numpy.float32'>")
+            self.assertEqual(err.record_name, keys[0])
+
+
+class TestBorealisConvert(unittest.TestCase):
+    """
+    Tests BorealisConvert class
+    """
+
+    def setUp(self):
+        pass
+
+    def test_borealis_convert_constructor(self):
+        """
+        Tests BorealisConvert constructor
+
+        Expected behaviour
+        ------------------
+        Contains file name of the data if given to it.
+        """
+        file_path = borealis_rawacf_file
+        converter = pydarn.BorealisConvert(file_path)
+        self.assertIsInstance(converter.group_names, list)
+        self.assertGreater(len(converter.group_names), 0)
+        self.assertEqual(converter.origin_filetype, 'rawacf')
+
+    def test_borealis_convert_to_rawacf(self):
+        """
+        Tests BorealisConvert to rawacf
+
+        Expected behaviour
+        ------------------
+        write a darn rawacf
+        """
+        file_path = borealis_rawacf_file
+        converter = pydarn.BorealisConvert(file_path)
+        converter.write_to_dmap("rawacf", "test_rawacf.rawacf.dmap")
+
+        # only testing the file is created since it should only be created
+        # at the last step after all checks have passed
+        # Testing the integrity of the insides of the file will be part of
+        # integration testing since we need BorealisRead for that.
+        self.assertTrue(os.path.isfile("test_rawacf.rawacf.dmap"))
+        os.remove("test_rawacf.rawacf.dmap")
+
+    def test_borealis_convert_to_iqdat(self):
+        """
+        Tests BorealisConvert to iqdat
+
+        Expected behaviour
+        ------------------
+        write a darn iqdat
+        """
+        file_path = borealis_bfiq_file
+        converter = pydarn.BorealisConvert(file_path)
+        converter.write_to_dmap("iqdat", "test_iqdat.iqdat.dmap")
+
+        # only testing the file is created since it should only be created
+        # at the last step after all checks have passed
+        # Testing the integrity of the insides of the file will be part of
+        # integration testing since we need BorealisRead for that.
+        self.assertTrue(os.path.isfile("test_iqdat.iqdat.dmap"))
+        os.remove("test_iqdat.iqdat.dmap")
```

Comment:
You can add a couple more tests with BorealisConvert in terms of trying to convert with the wrong files or cases where you cannot convert them because they are missing certain qualities. 

Also, there is a test/integration folder for integration tests, this is to test if you integrated class functionalities: 
BorealisConvert with DmapRead 
BorealisConvert with DARNRead
BorealisRead with BorealisWrite